api/proxy/README.md

```diff
@@ -283,7 +283,40 @@ $ set -a; source ./.env; set +a; ./bin/eigenda-proxy
 
 ### Features and Configuration Options (flags/env vars)
 
-Below is a list of the main high-level features offered for configuring the eigenda-proxy. These features are controlled via flags and/or env vars. To view the extensive list of available flags/env-vars to configure a given version of eigenda-proxy, run `eigenda-proxy --help`.
+Below is a list of the main high-level features offered for configuring the eigenda-proxy. These features are
+controlled via flags and/or env vars. To view the extensive list of available flags/env-vars to configure a given
+version of eigenda-proxy, run `eigenda-proxy --help`.
+
+#### Payment Mode Configuration
+
+When using EigenDA V2, the payment system can be configured using the `--eigenda.v2.client-ledger-mode` flag (or the
+`EIGENDA_PROXY_EIGENDA_V2_CLIENT_LEDGER_MODE` environment variable). This flag determines which payment mechanisms are
+active for blob dispersals. For detailed information about the payment system, see the
+[payment system documentation](../../docs/spec/src/protocol/payments/payment_system.md).
+
+**Available Payment Modes:**
+
+1. **`legacy` (default)** - Uses the legacy bin-based payment system that handles both reservation and on-demand
+   payments. This mode is in the process of being deprecated and will be removed in a future release.
+   
+> **IMPORTANT**: All clients should continue using this mode until the new payment system has officially shipped. The
+> other payment modes are documented below for awareness, but they aren't ready to be used in production environments.
+
+2. **`reservation-only`** - Uses pre-purchased bandwidth reservations that provide guaranteed throughput for a
+   specified time period. Bandwidth is managed using a leaky bucket algorithm. Dispersals will fail if a reservation
+   is temporarily exhausted.
+
+3. **`on-demand-only`** - Uses pay-per-dispersal payments from funds deposited in the PaymentVault contract.
+   Limited to quorums 0 (ETH) and 1 (EIGEN). Dispersals will fail if on-demand funds are exhausted.
+
+4. **`reservation-and-on-demand`** - Enables both reservation and on-demand payment methods with intelligent fallback.
+   Uses reservation bandwidth when available, and automatically switches to on-demand payments when reservation
+   capacity is temporarily exhausted. If a reservation *expires*, this mode will prevent any dispersals from being made
+   to avoid inadvertent draining of on-demand funds due to an expired reservation.
```

Comment:
nit: perhaps we should separate this into 2 modes? Something like "reservation-with-ondemand-excess" for this current mode and "reservation-and-on-demand" for literally allowing both.

Thinking there might be a team with a reservation expiring soon that wants to switch over to on-demand. Seems like don't currently provide a way to allow this smooth transition?

Reply:
I agree this makes sense to do.

I created a linear ticket to capture the idea, and will implement as a follow up at some point